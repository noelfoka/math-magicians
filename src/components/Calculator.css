body {
  margin: 6rem 0 0 20rem;
}

.calculator {
  display: grid;
  grid-template-rows: minmax(4rem, 1fr) repeat(5, 1fr);
  grid-template-columns: repeat(4, 5rem);
  padding: 5px;
}

.cal-answer {
  grid-column: 1 / span 4;
  background-color: rgba(179, 173, 173, 0.616);
  display: flex;
  flex-direction: column;
  align-items: flex-end;
  padding: 5px;
  margin-right: 1px;
}

.sign,
.assign {
  background-color: rgba(255, 115, 0, 0.94);
}

.sign:hover,
.assign:hover {
  background-color: rgba(255, 166, 0, 0.438);
}

.zero {
  grid-column: 1 / span 2;
  cursor: pointer !important;
}

button {
  margin: 1px;
  font-size: 1.3rem;
}
