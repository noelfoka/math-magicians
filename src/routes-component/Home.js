import React from 'react';
import styles from './Home.module.css';

function Home() {
  return (
    <div className={styles.home}>
      <h2 data-testid="user-id">History About Calculator</h2>
      <p>
        The earliest forms of calculators were mechanical devices, such as the
        abacus and slide rule. The abacus is a counting device that dates back
        to Roman times, while the slide rule appeared in the 17th century and
        was used to perform mathematical calculations. However, it was with the
        advent of electronics that modern calculators began to be developed. The
        first electronic calculators were created in the 1940s and 1950s, but
        they were very large and expensive, and were not suitable for consumer
        use. This changed in the 1960s when portable electronic calculators
        began to be marketed. The first consumer portable electronic calculator
        was the &rdquo;Cal Tech&rdquo;, launched in 1967 by Texas Instruments.
        It was capable of performing all four basic operations, but cost over
        $200, making it a luxury item at the time. Over the years, calculators
        have become more and more sophisticated and affordable. Scientific
        calculators appeared in the 1970s and were followed by graphing
        calculators in the 1980s. Nowadays, calculators are equipped with many
        advanced features such as graphing, statistics, and even programming. In
        summary, the calculator has come a long way since its mechanical
        beginnings centuries ago. Thanks to advances in technology, it has
        become an indispensable tool for students, professionals and scientists
        around the world.
      </p>
    </div>
  );
}

export default Home;
