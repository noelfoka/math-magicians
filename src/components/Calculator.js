import React from 'react';
import './Calculator.css';

const Calculator = () => (
  <div className="calculator">
    <div className="cal-answer">
      <div className="answer">0</div>
    </div>
    <button type="button" value="AC">
      AC
    </button>
    <button type="button" value="+/-">
      +/-
    </button>
    <button type="button" value="%">
      %
    </button>
    <button className="sign" type="button" value="/">
      /
    </button>
    <button type="button" value="7">
      7
    </button>
    <button type="button" value="8">
      8
    </button>
    <button type="button" value="9">
      9
    </button>
    <button className="sign" type="button" value="x">
      x
    </button>
    <button type="button" value="4">
      4
    </button>
    <button type="button" value="5">
      5
    </button>
    <button type="button" value="6">
      6
    </button>
    <button className="sign" type="button" value="-">
      -
    </button>
    <button type="button" value="1">
      1
    </button>
    <button type="button" value="2">
      2
    </button>
    <button type="button" value="3">
      3
    </button>
    <button className="sign" type="button" value="+">
      +
    </button>
    <button className="zero" type="button" value="0">
      0
    </button>
    <button type="button" value=".">
      .
    </button>
    <button className="assign" type="button" value="=">
      =
    </button>
  </div>
);

export default Calculator;
