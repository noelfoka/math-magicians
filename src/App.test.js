import { render, screen, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import userEvent from "@testing-library/user-event";
import Calculator from "./components/Calculator";
import Home from "./routes-component/Home";
import Quote from "./routes-component/Quotes";
import operate from "./logic/operate";
import calculate from "./logic/calculate";

afterEach(() => {
  cleanup();
});

describe("All components", () => {
  it("should render Home component", () => {
    render(<Home />);
    const userElement = screen.getByTestId("user-id");
    expect(userElement).toBeInTheDocument();
    expect(userElement).toHaveTextContent("History");
  });

  it("should render Calculator component", () => {
    render(<Calculator />);
    screen.getAllByRole("button");
  });

  it("should match HOME snapshot", () => {
    const home = renderer.create(<Home />).toJSON();
    expect(home).toMatchSnapshot();
  });

  it("should match QUOTE snapshot", () => {
    const quote = renderer.create(<Quote />).toJSON();
    expect(quote).toMatchSnapshot();
  });
});

describe("Conduct simple operations", () => {
  const number1 = 9;
  const number2 = 3;

  it("Click number to display", () => {
    render(<Calculator />);
    const number = screen.getByTestId("cal-answer");
    userEvent.click(screen.getByText("2"));
    expect(number.innerHTML).toMatch("2");
  });

  const addition = operate(number1, number2, "+");
  it("should perform an addition", () => {
    expect(addition).toBe("12");
  });

  const subtraction = operate(number2, number1, "-");
  it("should perform a subtraction", () => {
    expect(subtraction).toBe("-6");
  });

  const multiplication = operate(number1, number2, "x");
  it("should perform a multiplication", () => {
    expect(multiplication).toBe("27");
  });

  const division = operate(number1, number2, "รท");
  it("should perform a division", () => {
    expect(division).toBe("3");
  });

  const cantDivide = operate(number2, 0, "รท");
  it("should check for no divisible by 0", () => {
    expect(cantDivide).toBe("Can't divide by 0.");
  });

  const modulusOperator = operate(number1, number2, "%");
  it("should check for modulus of a number", () => {
    expect(modulusOperator).toBe("0");
  });
});
