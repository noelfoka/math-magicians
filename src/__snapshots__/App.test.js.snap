// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All components should match HOME snapshot 1`] = `
<div
  className="home"
>
  <h2
    data-testid="user-id"
  >
    History About Calculator
  </h2>
  <p>
    It is almost impossible for us to imagine mathematics without something as seemingly simple as a calculator. That being said, the calculators we know today were not invented until the 1970s, and the use of smartphones as calculators did not begin until at least the late nineties. But that doesn’t mean that mathematical tools were not available before the 20th century—there were many different computing machines created long before digital calculators and smartphones. The first tool created specifically for use in mathematical computations was the abacus, likely invented in Sumeria around 2500 B.C. The abacus was a table of successive columns with beads or stones representing a single unit, which could be used for addition or subtraction. Other cultures altered and refined the abacus; the Chinese, for example, put beads on wire within a bamboo frame to enhance ease of use. Unfortunately, the abacus was not very useful for multiplication or division, necessitating the invention of a new and more sophisticated device. Fast forward 4,500 years to 1617, when Scottish mathematician John Napier published Rabdology, or “calculation with rods.” In his writing, Napier described a device that came to be known as Napier’s bones. The “bones” are thin rods, which are inscribed with multiplication tables, and the user determines their sum by changing the vertical alignment of the rods, and horizontally reading the multiplication tables. While these devices greatly assisted calculations, they were not true “calculators,” merely assisting the person doing the mental calculation. In 1642, the first true “calculator” was invented: one that performed calculations through a clockwork-type of mechanism. The Pascal calculator, invented by French inventor and mathematician Blaise Pascal, was lauded for attempting arithmetic calculations previously thought impossible. But unfortunately, they were difficult to produce and very few were ever made. The mechanical calculator then invented by Thomas de Colmar in the mid-nineteenth century, and subsequent others, were easier to produce, extremely large and bulky–not at all the pocket calculators we know today. Curt Herzstark invented the first handheld, mechanic calculator in 1945, from a design he had created in 1938. Shaped like a stout pepper grinder, the Curta calculators were produced in large quantities until 1970, when a company in Japan invented the first digital pocket calculator and the demand for the antiquated device faded. Companies in the United States, such as Texas Instruments, adapted the design of the Japanese device, and enhanced it by creating the graphing calculators we know today. With the invention of the first smartphone in 1995, individuals began to replace expensive digital calculators with the multiuse device. This required even the most sophisticated calculator designs to be upgraded in order to remain relevant in the market.
  </p>
</div>
`;

exports[`All components should match QUOTE snapshot 1`] = `
<div
  className="quotes"
>
  <h1>
    I love this very motivational statement:
  </h1>
  <hr />
  <p>
    Loading...
  </p>
</div>
`;
